app/Makefile

```diff
@@ -33,15 +33,37 @@ else
 PY_RUN_CMD := docker-compose run $(DOCKER_EXEC_ARGS) --rm $(APP_NAME) poetry run
 endif
 
+# Docker user configuration
+#
+# Can be set by adding user=<username> and/ or uid=<id> to make command.
+#
+# If variables are not set explicitly: try looking up values from current
+# environment, otherwise fixed defaults.
+#
+# uid= defaults to 0 if user= set (which makes sense if user=root, otherwise you
+# probably want to set uid as well).
+#
+# Tested to work consistently on popular Linux flavors and Mac.
+ifeq ($(user),)
+RUN_USER ?= $(or $(strip $(USER)),nodummy)
+RUN_UID ?= $(or $(strip $(shell id -u)),4000)
+else
+RUN_USER = $(user)
+RUN_UID = $(or $(strip $(uid)),0)
+endif
+
+export RUN_USER
+export RUN_UID
+
 ##################################################
 # API Build & Run
 ##################################################
 
 build:
-	docker-compose build
+	docker-compose build $(APP_NAME)
 
 start:
-	docker-compose up --detach
+	docker-compose up --detach $(APP_NAME)
```

Comment:
Similar question, we need `main-db` to run too in order for the app to work, doesn't specifying $(APP_NAME) cause `main-db` not to run?

Reply:
No, in the `docker-compose.yml` file, we tell `main-app` that it is dependent on `main-db`, so it will build the DB first. I had this here for in case there are different docker-compose files. You could tell it to start the specified app, instead of it starting all apps

Reply:
Oh I get it thanks